### PR TITLE
Mass rename of networking tests

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
In order to make the tests more obvious to what they are actually testing rename the tests so that they fit the pattern

e2e-platform-networkPlugin-test-descroption

e2e-metal-ipi -> e2e-baremetal-ovn-ipv6
e2e-openstack -> e2e-openstack-ovn
e2e-windows-hybrid-network -> e2e-aws-ovn-windows
e2e-operator-with-custom-vxlan-port -> e2e-aws-ovn-windows-custom-vxlan

not sure what happpened to https://github.com/openshift/release/pull/12439 but this reflects the work that was there

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@dcbw I made the changes you talked about in https://github.com/openshift/release/pull/12439

**Comment by stbenjam** [comment]
Can you remove the baremetal changes from this PR? https://github.com/openshift/release/pull/12570 takes care of the renaming.  It also does some needed refactoring to make the jobs clearer.  The name also needs to keep the e2e-metal-ipi prefix, otherwise you'll break a lot of reporting automation outside of this repo (sippy, k8s testgrids, etc)

**Comment by danwinship** [comment]
> `rename e2e-aws -> e2e-aws-sdn in sdn repo`

OpenShift SDN in NetworkPolicy mode is the default for OCP, so tests that use that are not expected to indicate that fact, any more than they indicate "3 masters and 3 workers" or stuff like that. Even within repos where it might be ambiguous (eg, CNO), it's more important to have our naming match the naming of tests elsewhere in OCP than to have 100%-unambiguous names.

So, drop the "add -sdn" commits.

File: ci-operator/jobs/openshift/cluster-network-operator/openshift-cluster-network-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -712,11 +712,11 @@ presubmits:
         - name: CLUSTER_VARIANT
           value: ovn
         - name: JOB_NAME_SAFE
-          value: e2e-openstack
+          value: e2e-openstack-ovn
         - name: KURYR_ENABLED
           value: "false"
         - name: TEST_COMMAND
-          value: TEST_SUITE=openshift/conformance/parallel run-tests
+          value: TEST_SUITE=opensh  ift/conformance/parallel run-tests
```
Comment (dcbw) [inline ci-operator/jobs/openshift/cluster-network-operator/openshift-cluster-network-operator-master-presubmits.yaml:846]: @JacobTanenbaum this seems wrong...

Comment (JacobTanenbaum) [reply ci-operator/jobs/openshift/cluster-network-operator/openshift-cluster-network-operator-master-presubmits.yaml]: fixed

**Pull request merged**